src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/DockerRegistryManager.cs

```diff
@@ -20,11 +20,31 @@ public class DockerRegistryManager
     public const string FullyQualifiedBaseImageAspNet = $"{BaseImageSource}/{AspNetBaseImage}:{Net8PreviewImageTag}";
     private static string? s_registryContainerId;
 
-    public static void StartAndPopulateDockerRegistry(ITestOutputHelper testOutput)
+    private static string SDK_AzureContainerRegistryImage => "dotnetdhmirror-f8bzbjakh8cga6ab.azurecr.io/registry:2";
+    private static string Docker_HubRegistryImage => "docker.io/library/registry:2";
+
+    // TODO: some logic to pivot between this and Docker Hub
+    private static string RegistryImageToUse => SDK_AzureContainerRegistryImage;
+
+    internal class SameArchManifestPicker : IManifestPicker
+    {
+        public PlatformSpecificManifest? PickBestManifestForRid(IReadOnlyDictionary<string, PlatformSpecificManifest> manifestList, string runtimeIdentifier)
+        {
+            return manifestList.Values.SingleOrDefault(m => m.platform.os == "linux" && m.platform.architecture == "amd64");
+        }
+
+        public PlatformSpecificOciManifest? PickBestManifestForRid(IReadOnlyDictionary<string, PlatformSpecificOciManifest> manifestList, string runtimeIdentifier)
+        {
+            return manifestList.Values.SingleOrDefault(m => m.platform.os == "linux" && m.platform.architecture == "amd64");
+        }
+    }
```

Comment:
This part isn't used, and I think this Interface may not exist yet on this branch so it may lead to compilation errors


```suggestion
```